File: sample-app/src/main/java/com/globalpayments/android/sdk/sample/utils/FingerPrintUsageMethod.java
package com.globalpayments.android.sdk.sample.utils;

import static com.globalpayments.android.sdk.sample.common.Constants.FP_ALWAYS;
import static com.globalpayments.android.sdk.sample.common.Constants.FP_ON_SUCCESS;

import com.global.api.entities.Customer;

public class FingerPrintUsageMethod {

    public static Customer fingerPrintSelectedOption(FingerprintMethodUsageMode fingerprintMethodUsageMode) {
        Customer customer = new Customer();

        switch (fingerprintMethodUsageMode) {
            case ALWAYS:
                customer.setDeviceFingerPrint(FP_ALWAYS);
                break;
            case ON_SUCCESS:
                customer.setDeviceFingerPrint(FP_ON_SUCCESS);
                break;
        }
        return customer;
    }
}

File: sample-app/src/main/java/com/globalpayments/android/sdk/sample/utils/FingerPrintUsageMethod.kt
package com.globalpayments.android.sdk.sample.utils

import com.global.api.entities.Customer
import com.globalpayments.android.sdk.sample.common.Constants

object FingerPrintUsageMethod {
    @JvmStatic
    fun fingerPrintSelectedOption(fingerprintMethodUsageMode: FingerprintMethodUsageMode?): Customer {
        val customer = Customer()
        if (fingerprintMethodUsageMode != null) {
            when (fingerprintMethodUsageMode) {
                FingerprintMethodUsageMode.ALWAYS -> customer.deviceFingerPrint = Constants.FP_ALWAYS
                FingerprintMethodUsageMode.ON_SUCCESS -> customer.deviceFingerPrint = Constants.FP_ON_SUCCESS
            }
        }
        return customer
    }
}
